operator/controllers/keptntask/function_utils.go

```diff
@@ -0,0 +1,138 @@
+				"keptn.sh/version":  task.Spec.WorkloadVersion,
+			},
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy: "OnFailure",
+				},
+			},
+		},
+	}
+	err := controllerutil.SetControllerReference(task, job, r.Scheme)
+	if err != nil {
+		r.Log.Error(err, "could not set controller reference:")
+	}
+
+	container := corev1.Container{
+		Name:  "keptn-function-runner",
+		Image: "ghcr.io/keptn-sandbox/functions-runtime:main.202209211413",
+	}
+
+	var envVars []corev1.EnvVar
+
+	if len(params.Parameters) > 0 {
+		jsonParams, err := json.Marshal(params.Parameters)
+		if err != nil {
+			return job, fmt.Errorf("could not marshal parameters")
+		}
+		envVars = append(envVars, corev1.EnvVar{Name: "DATA", Value: string(jsonParams)})
+	}
+
+	if params.SecureParameters != "" {
+		envVars = append(envVars, corev1.EnvVar{
+			Name: "SECURE_DATA",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{Name: params.SecureParameters},
+					Key:                  "SECURE_DATA",
+				},
+			},
+		})
+	}
+
+	if params.ConfigMap != "" {
+		envVars = append(envVars, corev1.EnvVar{Name: "SCRIPT", Value: "/var/data/function.ts"})
+
+		job.Spec.Template.Spec.Volumes = []corev1.Volume{
+			{
+				Name: "function-mount",
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: params.ConfigMap,
+						},
+					},
+				},
+			},
+		}
+		container.VolumeMounts = []corev1.VolumeMount{
+			{
+				Name:      "function-mount",
+				ReadOnly:  true,
+				MountPath: "/var/data/function.ts",
+				SubPath:   "code",
+			},
+		}
+	} else {
+		envVars = append(envVars, corev1.EnvVar{Name: "SCRIPT", Value: params.URL})
+	}
+
+	container.Env = envVars
+	job.Spec.Template.Spec.Containers = []corev1.Container{
+		container,
+	}
+	return job, nil
+}
+
+func parseFunctionTaskDefinition(definition *klcv1alpha1.KeptnTaskDefinition) (FunctionExecutionParams, bool, error) {
+	params := FunctionExecutionParams{}
+
+	// Firstly check if this task definition has a parent object
+	hasParent := false
+	if definition.Spec.Function.FunctionReference != (klcv1alpha1.FunctionReference{}) {
+		hasParent = true
+	}
+
+	// Check if there is a ConfigMap with the function for this object
+	if definition.Status.Function.ConfigMap != "" {
+		params.ConfigMap = definition.Status.Function.ConfigMap
+	} else {
+		// If not, check if it has an HTTP reference. If this is also not the case and the object has no parent, something is wrong
+		if definition.Spec.Function.HttpReference.Url == "" && !hasParent {
+			return params, false, fmt.Errorf("No ConfigMap specified or HTTP source specified in TaskDefinition) / Namespace: %s, Name: %s ", definition.Namespace, definition.Name)
+		}
+		params.URL = definition.Spec.Function.HttpReference.Url
+	}
+
```

Comment:
should we throw an error if we have both URL and CM? 
I think having CM as a winner without error is the best option tho

Reply:
added